draw/DrawingProgram.py

```diff
@@ -0,0 +1,77 @@
+# Your team will create a program that has a class that holds a list/collection of shapes. This class will
+# be named DrawingProgram. Here are the attributes (data) and behaviors (methods) of this class:
+from shape_factory import ShapeFactory
+
+
+class DrawingProgram(ShapeFactory):
+    # For the intents and purposes of this assignment, all shapes should be created via ShapeFactory.
+    #     Thus any adding of shapes to your DrawingProgram class object must be done through ShapeFactory.
+    #     This may seem cumbersome, but the point is to leave object creation to a single class.
+    #         It encapsulates how to create the objects you need.
+    #  ^^^ I think this means we pass in ShapeFactory into DrawingProgram instead of the shapes themselves?
+
+    def __init__(self, shapes = None):
+        if shapes is None or isinstance(shapes, list):
```

Comment:
We need it initiate `self.shapes` as an empty list for other things to work down stream. For example if we try use `.append()` when self.shapes is `None` we will get an error. 
I'm going to fix this as I write the `DrawingProgramIterator`. I'll make it an empyt list by default.